docs/building.md

```diff
@@ -0,0 +1,125 @@
+# Akri Build
+Building Akri, whether locally or in the automated CI builds, leverages the same set of `make` commands.
+
+In essence, there are 3 types of builds for Akri:
+1. Core components 
+    1. Rust code: containers based on Rust code are built using `Cargo cross` and subsequent `docker build` commands include the cross-built binaries.
+        > Note: For Rust code, build/Dockerfile.* does NOT run `cargo build`, instead they simply copy cross-built binaries into the container
+    2. Other code: these containers can be .NET or python or whatever else ... the build/Dockerfile.* must do whatever building is required.
+2. Intermediate builds: these containers are used as part of the build process and are not used in production explicitly
+
+## Core components
+The core Akri components are the containers that are downloaded at runtime to provide Akri's functionality.  They include the agent, the controller, the webhook,  the brokers, and the applications.  The majority of Akri component code is written in Rust, but there are several components that are written in .NET or python.
```

Comment:
Should the brokers and applications be referred to as Core Akri components?

Reply:
probably not, but i wasn't sure how to describe not-intermediate things ... is "Akri runtime components" better?

Reply:
I was wondering if Akri's core components are really just control, agent, webhook. And the rest are addons and demos.

Reply:
yes, i think your description is accurate.